x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_logic.ts

```diff
@@ -0,0 +1,271 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { kea } from 'kea';
+
+import { formatApiName } from '../../utils';
+import { ADMIN, PRIVATE } from '../../constants/credentials';
+
+import { HttpLogic } from '../../../shared/http';
+import { IApiToken, IEngine, ICredentialsDetails } from '../../../../../common/types/app_search';
+import { IMeta } from '../../../../../common/types';
+
+const defaultApiToken = {
+  name: '',
+  type: PRIVATE,
+  read: true,
+  write: true,
+  access_all_engines: true,
+};
+
+// TODO CREATE_MESSAGE, UPDATE_MESSAGE, and DELETE_MESSAGE from ent-search
```

Comment:
There are a number of things that still need to be ported over. The things that were not ported over were not necessary for the credentials list, and have a dependency on roles, which are not yet available in AppLogic.

I plan to add these in a subsequent PR, when and as they are needed and available.

Reply:
Is this a useful TODO to include in open source code? Maybe better off to create an issue and leave this note out.